### PR TITLE
Add lifecycle prevent destroy to Secure Source Manager examples

### PR DESCRIPTION
**Pull request by ericpang777**
**Release Note Template for Downstream PRs (will be copied)**

```release-note:none
add lifecycle prevent destroy to Secure Source Manager instance and repository examples
```

### CODE REVIEW

**Comment by ericpang777** [comment]
Is there any need to add oics_var_overrides? I can't tell what it is used for but some examples seem to use it...

File: mmv1/templates/terraform/examples/secure_source_manager_instance_basic.tf.erb [hunk 1]
```diff
@@ -4,4 +4,8 @@ resource "google_secure_source_manager_instance" "<%= ctx[:primary_resource_id]
     labels = {
       "foo" = "bar"
     }
+
+    lifecycle {
```
Comment (umairidris) [inline mmv1/templates/terraform/examples/secure_source_manager_instance_basic.tf.erb:8]: Should we add a comment explaining why we are adding this block?

Comment (ericpang777) [reply mmv1/templates/terraform/examples/secure_source_manager_instance_basic.tf.erb:9]: Added.

**Review by slevenick — APPROVED** [review]
Seems fine to me as a docs-only change

**Comment by slevenick** [comment]
@umairidris ping me when you are good with this and I'll merge it

**Comment by ericpang777** [comment]
> Seems fine to me as a docs-only change

Hi Sam, do I need to add `oics_var_overrides`? I notice some tests that have `test_var_overrides` also use it, but I have no idea what it does.

File: mmv1/templates/terraform/examples/secure_source_manager_instance_cmek.tf.erb [hunk 2]
```diff
@@ -23,6 +23,11 @@ resource "google_secure_source_manager_instance" "<%= ctx[:primary_resource_id]
     depends_on = [
       google_kms_crypto_key_iam_member.crypto_key_binding
     ]
+
+    # For preventing accidental deletions
```
Comment (umairidris) [inline mmv1/templates/terraform/examples/secure_source_manager_instance_cmek.tf.erb:27]: Nit:

Prevent accidental deletions.

(full sentence ending with period)

Comment (ericpang777) [reply mmv1/templates/terraform/examples/secure_source_manager_instance_cmek.tf.erb]: Done

**Comment by slevenick** [comment]
> > Seems fine to me as a docs-only change
> 
> Hi @slevenick, do I need to add `oics_var_overrides`? I notice some tests that have `test_var_overrides` also use it, but I have no idea what it does.

Hmmm, probably not? oics is open in cloud shell, but I don't think you would need specific overrides for that?

**Comment by slevenick** [comment]
> > > Seems fine to me as a docs-only change
> > 
> > 
> > Hi @slevenick, do I need to add `oics_var_overrides`? I notice some tests that have `test_var_overrides` also use it, but I have no idea what it does.
> 
> Hmmm, probably not? oics is open in cloud shell, but I don't think you would need specific overrides for that?

Well, actually thinking about the use case if someone creates one of these in cloud shell it's probably just for testing purposes and they'll want to delete it. So it may be good to include the same overrides for oics as are used in tests

**Review by umairidris — APPROVED** [review]

**Comment by ericpang777** [comment]
> > > > Seems fine to me as a docs-only change
> > > 
> > > 
> > > Hi @slevenick, do I need to add `oics_var_overrides`? I notice some tests that have `test_var_overrides` also use it, but I have no idea what it does.
> > 
> > 
> > Hmmm, probably not? oics is open in cloud shell, but I don't think you would need specific overrides for that?
> 
> Well, actually thinking about the use case if someone creates one of these in cloud shell it's probably just for testing purposes and they'll want to delete it. So it may be good to include the same overrides for oics as are used in tests

Mhm that's a good point, added 👍

**Review by umairidris — APPROVED** [review]

**Pull request merged**